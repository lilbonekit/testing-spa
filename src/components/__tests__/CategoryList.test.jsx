import { screen } from '@testing-library/react';
import { renderWithRouter } from '../../utils/testing';

import { CategoryList } from '../CategoryList';

describe('CategoryList', () => {
  it('renders correctly', () => {
    renderWithRouter(
      <CategoryList
        catalog={[
          {
            idCategory: '1',
            strCategory: 'Beef',
            strCategoryThumb: 'themodi.png',
            strCategoryDescription: 'strCategoryDescription',
          },
        ]}
      />,
    );
    expect(screen.getByRole('list')).toMatchSnapshot();
  });

  it('renders correctly with no items', () => {
    renderWithRouter(<CategoryList />);
    expect(screen.getByRole('list')).toMatchSnapshot();
  });
});
