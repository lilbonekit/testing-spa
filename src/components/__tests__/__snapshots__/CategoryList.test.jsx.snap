// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList renders correctly 1`] = `
<div
  class="list"
  role="list"
>
  <div
    class="card"
    role="article"
  >
    <div
      class="card-image"
    >
      <img
        alt="Beef"
        src="themodi.png"
      />
    </div>
    <div
      class="card-content"
    >
      <span
        class="card-title"
      >
        Beef
      </span>
      <p>
        strCategoryDescription
        ...
      </p>
    </div>
    <div
      class="card-action"
    >
      <a
        class="btn"
        href="/category/Beef"
      >
        Watch category
      </a>
    </div>
  </div>
</div>
`;

exports[`CategoryList renders correctly with no items 1`] = `
<div
  class="list"
  role="list"
/>
`;
